fixedpointmath/fixed_point_math.py

```diff
@@ -66,32 +66,40 @@ def isclose(a: NUMERIC, b: NUMERIC, abs_tol: NUMERIC = FixedPoint("0.0")) -> boo
     return abs(a - b) <= abs_tol
 
 
-def maximum(x: NUMERIC, y: NUMERIC) -> NUMERIC:
-    """Compare the two inputs and return the greater value.
+def maximum(*args: NUMERIC) -> NUMERIC:
+    """Compare the inputs and return the greatest value.
 
     If the first argument equals the second, return the first.
     """
-    if isinstance(x, FixedPoint) and x.is_nan():
-        return x
-    if isinstance(y, FixedPoint) and y.is_nan():
-        return y
-    if x >= y:
-        return x
-    return y
-
-
-def minimum(x: NUMERIC, y: NUMERIC) -> NUMERIC:
-    """Compare the two inputs and return the lesser value.
+    # use builtin for generic types
+    if isinstance(args[0], (float, int)):
```

Comment:
Is this desired? If I pass in `maximum(0, FixedPoint(1)), it'll return 1?

Reply:
Also, add check for length of args. Right now, if I call `maximum()`, it will complain at this line for invalid index

Reply:
> Is this desired? If I pass in `maximum(0, FixedPoint(1)), it'll return 1?

this is not allowed -- all args must be the same type. This was enforced via types but not enforced any other way. It's still not strictly enforced, but more so by passing off to the stdlib fn.

Reply:
> Also, add check for length of args. Right now, if I call `maximum()`, it will complain at this line for invalid index

good catch! on it.